exporter.go

```diff
@@ -114,33 +114,56 @@ func (e *Exporter) Export(layersDir, appDir string, runImage, origImage imgutil.
 	if err != nil {
 		return errors.Wrap(err, "marshall metadata")
 	}
-	if err := appImage.SetLabel(metadata.AppMetadataLabel, string(data)); err != nil {
+
+	if err = workingImage.SetLabel(metadata.AppMetadataLabel, string(data)); err != nil {
 		return errors.Wrap(err, "set app image metadata label")
 	}
 
-	if err := appImage.SetEnv(cmd.EnvLayersDir, layersDir); err != nil {
+	if err = workingImage.SetEnv(cmd.EnvLayersDir, layersDir); err != nil {
 		return errors.Wrapf(err, "set app image env %s", cmd.EnvLayersDir)
 	}
 
-	if err := appImage.SetEnv(cmd.EnvAppDir, appDir); err != nil {
+	if err = workingImage.SetEnv(cmd.EnvAppDir, appDir); err != nil {
 		return errors.Wrapf(err, "set app image env %s", cmd.EnvAppDir)
 	}
 
-	if err := appImage.SetEntrypoint(launcher); err != nil {
+	if err = workingImage.SetEntrypoint(launcher); err != nil {
 		return errors.Wrap(err, "setting entrypoint")
 	}
 
-	if err := appImage.SetCmd(); err != nil { // Note: Command intentionally empty
+	if err = workingImage.SetCmd(); err != nil { // Note: Command intentionally empty
 		return errors.Wrap(err, "setting cmd")
 	}
 
-	sha, err := appImage.Save()
-	if err != nil {
-		return errors.Wrap(err, "saving")
+	err = workingImage.Save(additionalNames...)
+	if _, isSaveErr := err.(imgutil.SaveError); err != nil && !isSaveErr {
+		return errors.Wrap(err, "saving image")
+	}
+
+	digest, digestErr := workingImage.Digest()
+	if digestErr != nil {
+		return errors.Wrap(digestErr, "getting digest")
 	}
-	e.Out.Printf("\n*** Image: %s@%s\n", runImage.Name(), sha)
+	e.Out.Printf("\n*** Digest: %s\n", digest)
 
-	return nil
+	e.Out.Println("*** Images:")
+	for _, n := range append([]string{workingImage.Name()}, additionalNames...) {
+		e.Out.Printf("      %s - %s\n", n, getSaveStatus(err, n))
```

Comment:
I think we should avoid reinterpreting this error again later. Let's extract what we need if the first error handling block.